src/api/ppx_deriving.cppo.ml

```diff
@@ -17,28 +8,122 @@
                                                { txt = txt_; loc = loc_ }; \
                                               attr_payload = payload; \
                                               attr_loc = _ }
-#endif
-
-#if OCAML_VERSION < (4, 08, 0)
-#define Rtag_patt(label, constant, args) Rtag(label, _, constant, args)
-#define Rinherit_patt(typ) Rinherit(typ)
-#else
-#define Rtag_patt(label, constant, args) {prf_desc = Rtag(label, constant, args); _}
-#define Rinherit_patt(typ) {prf_desc = Rinherit(typ); _}
-#endif
-
-open Longident
 open Location
 open Asttypes
-open Parsetree
 open Ast_helper
+
+module Ast_convenience = struct
```

Comment:
Do we really need to redefine this? As far as I see, we are upgrading the version of the AST used by ppx_deriving - that is already a breaking change. We might as well ask users to switch to the ppxlib API in their plugins as well.

Reply:
The idea was mainly to reduce the amount of changes elsewhere in the library (because `Ast_convenience` is used a lot of times in standard plugins).